File: Python3/1-100/1-two-sum.py
#1. Two Sum (Python3) - 56ms/88.22% 15.6MB/9.74%
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        halves = dict()
        for i, val in enumerate(nums):
            if val in halves:
                return [halves.get(val), i]
            halves[target-val] = i

File: C++/1-100/1-two-sum.cpp
//#1. Two Sum (C++) - 11ms/90.86% 11MB/31.48%
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> match;
        for (int i = 0; i < nums.size(); i++) {
            int partner = target - nums[i];
            auto query = match.find(partner);
            if (query != match.end()) {
                vector<int> results{query->second, i};
                return results;
            }
            match.emplace(nums[i], i);
        }
        vector<int> out;
        return out;
    }
};